sdcm/sct_runner.py

```diff
@@ -253,96 +378,80 @@ def _create_instance(self, instance_type, base_image, tags_list, instance_name=N
                 "Groups": [aws_region.sct_security_group.group_id],
                 "DeleteOnTermination": True,
             }],
-            TagSpecifications=[{"ResourceType": "instance", "Tags": tags_list}],
+            TagSpecifications=[{
+                "ResourceType": "instance",
+                "Tags": [{"Key": key, "Value": value} for key, value in tags.items()] +
+                        [{"Key": "Name", "Value": instance_name}],
+            }],
             BlockDeviceMappings=[{
-                "DeviceName": ec2_ami_get_root_device_name(image_id=base_image, region=region),
+                "DeviceName": ec2_ami_get_root_device_name(image_id=base_image, region=aws_region.region_name),
                 "Ebs": {
                     "VolumeSize": self.instance_root_disk_size(test_duration),
                     "VolumeType": "gp2"
                 }
             }]
         )
         instance = result[0]
+
         LOGGER.info("Instance created. Waiting until it becomes running... ")
         instance.wait_until_running()
-        LOGGER.info("Instance '%s' is running. Waiting for public IP...", instance.instance_id)
+
+        LOGGER.info("Instance `%s' is running. Waiting for public IP...", instance.instance_id)
         ec2_instance_wait_public_ip(instance=instance)
+
         LOGGER.info("Got public IP: %s", instance.public_ip_address)
+
         return instance
 
+    def _stop_image_builder_instance(self, instance: Any) -> None:
+        instance.stop()
+        instance.wait_until_stopped()
+
+    def _terminate_image_builder_instance(self, instance: Any) -> None:
+        instance.terminate()
+
+    def _get_instance_id(self, instance: Any) -> Any:
+        return instance.instance_id
+
+    def get_instance_public_ip(self, instance: Any) -> str:
+        return instance.public_ip_address
+
     def tag_image(self, image_id, image_type) -> None:
         if image_type == ImageType.SOURCE:
-            ec2_resource = self.ec2_resource_source
+            aws_region = self.aws_region_source
         elif image_type == ImageType.GENERAL:
-            ec2_resource = self.ec2_resource
-        image_tags = [
-            {"Key": "Name", "Value": self.image_name},
-            {"Key": "Version", "Value": str(self.VERSION)},
-        ]
-        runer_image = ec2_resource.Image(image_id)
-        runer_image.wait_until_exists()
+            aws_region = self.aws_region
+        else:
+            raise ValueError("Unknown Image type")
```

Comment:
Would be useful to add the value that is "unknown". In many cases is eases debugging when something is wrong.